### ISSUE TITLE
Changed behavior when old network is used: losing game with passes

### ISSUE THREAD

**Issue by pliuphys** [leela-zero/leela-zero#1879]
I'm trying to have leela networks play each other and get selfplay elo. I chose to run 60 vs 70 first. With gogui-twogtp, I got some games where 70 lost because of passing, not taking dead stones. I looked at about a dozen counted games LZ70 played in the official matches but didn't see any of this. Removing -d makes the bot keep playing, but the official matches had -d. This happens perhaps a couple times per 100 games.

I downloaded version 0.11 which the matches used (I was running 0.15) and for the second pass it wasn't in the first few moves (by visits, I guess) for 0.11. With fpu_reduction set to 0 in 0.15 white continues to play. In one test with 1600 visits the top move p19 gets 136 visits, pass gets 53, at number 8.

I guess to test the old networks I'll have to use the versions where the official matches were made.

=================
Excerpt from log file:

W<< = pass
W<< 
B>> play W PASS
B<< = 
B<< 
B>> genmove b
Thinking at most 87.4 seconds...
NN eval=0.140900
86.5s left, stopping early.

 D13 ->     265 (V: 12.15%) (N: 12.69%) PV: D13 P19 B11 L19 M11 pass M9 pass N10 pass C10 pass O9 pass P13 pass M14 pass
  O9 ->     190 (V: 13.37%) (N:  1.77%) PV: O9 pass D13 P19 B11 L19 M9 pass N10 pass M11 E1 C10 pass M14 pass A12 pass A10
 M15 ->     165 (V: 13.44%) (N:  1.42%) PV: M15 P19 D13 L19 B11 pass M9 pass M11 pass N10 pass C10 pass O14 pass O9 pass A12 pass D7 pass
 B11 ->      96 (V: 12.37%) (N:  4.94%) PV: B11 pass D13 L19 C10 E1 M9 P19 M11 pass N10 pass O9 pass M14 pass A12
 O10 ->      84 (V: 13.43%) (N:  1.23%) PV: O10 Q16 D13 P19 B11 L19 N11 pass M9 pass C10
  F1 ->      72 (V: 13.19%) (N:  1.60%) PV: F1 E1 D13 L19 B11 P19 C10 pass M9 pass M11 pass N10 pass P13 pass O9
 N11 ->      72 (V: 13.02%) (N:  1.99%) PV: N11 pass D13 pass B11 P19 M9 pass C10 pass O14 pass O9 pass N10
 M11 ->      62 (V: 12.76%) (N:  2.49%) PV: M11 pass D13 L19 M9 P19 B11 pass N10
 C10 ->      60 (V: 12.79%) (N:  2.28%) PV: C10 pass D13 pass B11 L19 M9 P19 N10 pass M11 pass O9 pass P13 pass M14 pass D7 pass
 E13 ->      58 (V: 12.91%) (N:  2.08%) PV: E13 A19 D13 L19 B11 P19 C10 pass M9 pass N10 pass
 P13 ->      57 (V: 12.89%) (N:  2.05%) PV: P13 L19 D13 P19 B11 pass M9 pass M11 pass N10 pass C10 pass M14 pass D7
  M9 ->      52 (V: 12.24%) (N:  2.71%) PV: M9 P19 D13 L19 B11 pass M11 pass N10 pass C10 pass O9 pass P13 pass M14 pass
 O14 ->      36 (V: 12.43%) (N:  1.87%) PV: O14 P19 D13 L19 B11 pass M11 pass M9 pass N10 pass C10
  S1 ->      30 (V: 12.01%) (N:  1.71%) PV: S1 L19 D13 P19 B11 pass M9 pass C10 pass N10 pass M11 pass O14
 C11 ->      27 (V: 12.12%) (N:  1.70%) PV: C11 pass D13 L19 M9 P19 M11 pass N10 pass O9 pass O14 pass B11 pass M14 pass A12 pass D8 pass A10 pass S16 S17 Q16
  A1 ->      26 (V: 12.37%) (N:  1.13%) PV: A1 A2 D13 pass B11 L19 M9 pass C10 pass M11 P19 N10 pass P13
 S16 ->      24 (V: 12.30%) (N:  1.32%) PV: S16 S17 D13 L19 B11 P19 M9 pass M11 pass N10 pass C10 pass O14 pass T19
 M12 ->      24 (V: 12.07%) (N:  1.49%) PV: M12 L19 D13 P19 B11 pass M9 pass N10 pass C10 pass O9 pass O14 pass M11 pass A12 pass
  D7 ->      19 (V: 11.42%) (N:  1.64%) PV: D7 pass D13 P19 B11 pass C10 pass M9 pass M11
 M14 ->      19 (V: 11.01%) (N:  1.99%) PV: M14 P19 D13 L19 B11 pass M9 pass N10 pass M11 pass C10
 N10 ->      19 (V: 10.77%) (N:  1.95%) PV: N10 L19 D13 P19 B11 pass M9 pass M11 pass C10 pass O9
 M13 ->      18 (V: 11.81%) (N:  1.22%) PV: M13 P19 D13 L19 B11 pass M11 pass M9 pass N10 pass C10 pass
 O13 ->      16 (V: 11.53%) (N:  1.21%) PV: O13 pass D13 L19 B11 P19 M9 pass N10 pass
  D8 ->      16 (V: 11.38%) (N:  1.32%) PV: D8 E1 D13 L19 B11 pass M9 pass
 B10 ->      12 (V: 10.75%) (N:  1.26%) PV: B10 pass D13 pass M9 pass N10 P19 M11 pass
  C7 ->       6 (V:  6.20%) (N:  1.39%) PV: C7 pass D13 pass B11
8.5 average depth, 28 max depth
1244 non leaf nodes, 1.23 average children
1526 visits, 134854 nodes, 1490 playouts, 1568 n/s

B<< = D13
B<< 
W>> play B D13
W<< = 
W<< 
W>> genmove w
Thinking at most 87.3 seconds...
NN eval=0.983824
86.5s left, stopping early.

pass ->     244 (V: 99.90%) (N:  9.74%) PV: pass P13 pass B11 pass M14 pass M9 pass
 L19 ->     242 (V: 99.90%) (N:  9.69%) PV: L19 P13 P19 B11 pass M14 pass M9 pass
 Q16 ->     162 (V: 99.90%) (N:  7.28%) PV: Q16 P13 pass B11 pass M14 pass
 P19 ->     140 (V: 99.89%) (N:  6.49%) PV: P19 B11 pass P13 pass M9 pass M11
  A1 ->      99 (V: 99.90%) (N:  4.82%) PV: A1 B11 pass P13 pass M9 L19
 T19 ->      71 (V: 99.90%) (N:  3.55%) PV: T19 P13 P19 S19 R19 B11
 A19 ->      64 (V: 99.90%) (N:  3.21%) PV: A19 P13 pass M9 L19 M11
 Q17 ->      53 (V: 99.90%) (N:  2.72%) PV: Q17 P13 pass B11 P19 M14
 C16 ->      45 (V: 99.90%) (N:  2.30%) PV: C16 B11 pass P13 L19 M14 pass
  A2 ->      44 (V: 99.89%) (N:  2.26%) PV: A2 P13 pass M9 pass B11
 C17 ->      40 (V: 99.89%) (N:  2.07%) PV: C17 B11 pass P13 pass C10 pass
 A18 ->      40 (V: 99.89%) (N:  2.07%) PV: A18 P13 L19 B11 pass M9
  E1 ->      32 (V: 96.78%) (N:  5.07%) PV: E1 pass Q16 P13 P19 B11 pass
 N18 ->      31 (V: 99.90%) (N:  1.64%) PV: N18 P13 pass B11 pass
 K17 ->      30 (V: 99.90%) (N:  1.59%) PV: K17 P13 L19 B11 pass
  C3 ->      30 (V: 99.89%) (N:  1.60%) PV: C3 B11 pass P13 L19 M14
 Q19 ->      29 (V: 99.90%) (N:  1.54%) PV: Q19 P13 L19 B11 pass
 S19 ->      28 (V: 99.90%) (N:  1.48%) PV: S19 P13 L19 M9 pass
 O19 ->      27 (V: 99.89%) (N:  1.45%) PV: O19 P13 L19 M9 pass B11
 R19 ->      21 (V: 99.90%) (N:  1.13%) PV: R19 P13 L19 B11 pass
 O14 ->      20 (V: 99.88%) (N:  1.09%) PV: O14 P13 pass O13 L19 B11
 K16 ->      19 (V: 99.90%) (N:  0.99%) PV: K16 B11 pass P13 pass M9 pass
 S16 ->      19 (V: 99.88%) (N:  1.03%) PV: S16 P13 L19 B11 pass M14
 P13 ->      19 (V: 99.87%) (N:  1.01%) PV: P13 O14 L19 O13 pass B11 pass
 S17 ->      18 (V: 99.88%) (N:  0.97%) PV: S17 O14 pass B11 pass
  F1 ->      16 (V: 99.90%) (N:  0.85%) PV: F1 P13 P19 B11 pass
 T18 ->      15 (V: 99.90%) (N:  0.83%) PV: T18 O14 P19 B11 pass
5.1 average depth, 10 max depth
1164 non leaf nodes, 1.37 average children
1599 visits, 144567 nodes, 1533 playouts, 1722 n/s

W<< = pass
W<< 
B>> play W PASS
B<< = 
B<< 
B>> genmove b
Thinking at most 87.4 seconds...
NN eval=0.125318
87.1s left, stopping early.

pass ->     479 (V: 100.00%) (N:  1.25%) PV: pass 
 C10 ->      95 (V: 12.24%) (N:  3.21%) PV: C10 pass B11 L19 M9 P19 N10 pass M11 pass O9 pass P13 pass M14 pass D7 pass
 B11 ->      91 (V: 11.46%) (N:  6.51%) PV: B11 L19 C10 E1 M9 P19 M11 pass N10 pass O9 pass M14 pass A12
 M15 ->      84 (V: 12.59%) (N:  1.51%) PV: M15 pass B11 pass M9 L19 C10 P19 N10 pass M11 pass O14 pass O9 pass A12 pass
 O10 ->      63 (V: 12.35%) (N:  2.06%) PV: O10 P19 B11 T19 N11 L19 M9 pass C10 pass M12 pass S19 R19 S16 S17 Q16
  O9 ->      56 (V: 11.91%) (N:  2.66%) PV: O9 P19 B11 L19 M9 pass N10 pass M11 E1 C10 pass M14 pass A12 pass A10
 P13 ->      48 (V: 12.12%) (N:  2.08%) PV: P13 pass B11 P19 M9 pass M11 L19 N10 pass C10 pass M14
 O14 ->      44 (V: 11.99%) (N:  1.76%) PV: O14 pass B11 pass C10 pass M9 pass N10 L19 M11
  F1 ->      39 (V: 11.45%) (N:  2.80%) PV: F1 E1 B11 pass C10 pass M9 pass N10 pass M11 P19 O14 pass O9 pass M14 pass D7 pass A12
  M9 ->      39 (V: 11.25%) (N:  3.00%) PV: M9 L19 B11 pass M11 pass N10 P19
 N10 ->      33 (V: 10.98%) (N:  2.81%) PV: N10 pass B11 pass M9 pass M11 pass C10 pass O14 L19 D7
 A12 ->      18 (V: 11.07%) (N:  1.54%) PV: A12 P19 B11 pass M9 pass N10 L19 M11
  D8 ->      16 (V: 10.33%) (N:  1.92%) PV: D8 pass B11 pass M9 P19 M11 pass N10 pass C10 pass O9
 N11 ->      16 (V: 10.04%) (N:  2.27%) PV: N11 pass B11 P19 M9 pass C10 pass O14 pass O9 pass N10
 M12 ->      15 (V: 10.94%) (N:  1.38%) PV: M12 pass B11 L19 M9 P19 N11 pass C10 pass O14 pass O10 pass F1
  S1 ->      14 (V: 10.65%) (N:  1.52%) PV: S1 P19 B11 L19 M9 pass C10 pass N10 pass
 O13 ->      14 (V: 10.58%) (N:  1.55%) PV: O13 L19 B11 P19 M9 pass N10 pass
 M11 ->      14 (V:  9.26%) (N:  2.49%) PV: M11 L19 M9 P19 B11 pass N10
 M14 ->      13 (V: 10.45%) (N:  1.53%) PV: M14 P19 B11 L19 M9 pass N10 pass
  C7 ->      10 (V:  9.66%) (N:  1.49%) PV: C7 pass B11 pass M9 P19 M11
 B10 ->       8 (V:  9.57%) (N:  1.34%) PV: B10 pass M9 pass N10 P19 M11 pass
  D7 ->       8 (V:  7.96%) (N:  2.01%) PV: D7 P19 B11 pass C10 pass M9 pass
7.5 average depth, 22 max depth
604 non leaf nodes, 1.22 average children
1218 visits, 65230 nodes, 1180 playouts, 3933 n/s

B<< = pass
B<< 
W>> play B PASS
W<< = 
W<< 
B>> final_score
B<< = B+5.5
B<< 
W>> final_score
W<< = B+5.5
W<< 

[60_70_r1-56.zip](https://github.com/gcp/leela-zero/files/2418718/60_70_r1-56.zip)

**Comment by gcp** [comment]
Don't think you can conclude anything about this with just a few games or looking at just a few positions. Leela had passing problems for a long time until the nets scored it very confidently, and that's not really related to changing search parameters. Changing search parameters randomly in a position where it played the wrong move often makes it play a better move...and this is completely meaningless. We've covered this many times before.

**Comment by pliuphys** [comment]
@gcp What you said is completely true. When I looked harder I found one in the official match with 70 vs 69 http://zero.sjeng.org/viewmatch/0b28cc082a2705c083324a907bee748ed0c27a0ce2fb8b29d40e021f7a2d9d47?viewer=wgo

I'm running 60(0.15 default) vs 70(0.15 fpu=0) and 70 has lost a game with bad pass. It also won a smaller fraction of the games.

I guess I can use the current version to continue testing the old networks. I thought there might be something wrong in my configuration when I first saw these.
